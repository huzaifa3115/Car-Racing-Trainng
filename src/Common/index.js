import Styles from './Styles';
import Data from './Data';

export {Styles, Data};
