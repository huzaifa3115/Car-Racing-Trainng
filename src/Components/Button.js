import * as React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';

interface Props {
  label: string;
  onPress: () => void;
  btnStyle: {};
  labelStyle: {};
}

class Button extends React.Component<Props> {
  render() {
    const {label, onPress, labelStyle, btnStyle} = this.props;
    return (
      <TouchableOpacity style={[styles.container, btnStyle]} onPress={onPress}>
        <Text style={[styles.text, labelStyle]}>{label}</Text>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#428AF8',
    marginBottom: 12,
    paddingVertical: 12,
    borderRadius: 4,
    borderWidth: StyleSheet.hairlineWidth,
    borderColor: 'rgba(255,255,255,0.7)',
  },
  text: {
    color: '#fff',
    textAlign: 'center',
    fontSize: 16,
  },
});

export default Button;
