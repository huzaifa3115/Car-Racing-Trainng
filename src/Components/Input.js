import * as React from 'react';
import {StyleSheet, TextInput, TextInputProps} from 'react-native';

type Props = TextInputProps;

class Input extends React.Component<Props> {
  render() {
    const {style, ...otherProps} = this.props;
    return (
      <TextInput
        selectionColor={'#428AF8'}
        style={[styles.textInput, style]}
        {...otherProps}
      />
    );
  }
}

const styles = StyleSheet.create({
  textInput: {
    height: 40,
    borderColor: '#BEBEBE',
    borderBottomWidth: StyleSheet.hairlineWidth,
    marginBottom: 20,
  },
});

export default Input;
