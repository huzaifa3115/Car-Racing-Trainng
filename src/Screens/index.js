import AppLoading from './AppLoading';
import Login from './Login';
import Register from './Register';
import Track from './Track';

export {AppLoading, Login, Register, Track};
