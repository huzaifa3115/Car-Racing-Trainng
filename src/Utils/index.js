import Utils from './Utils';
import Navigator from './Navigator';

export {Utils, Navigator};
