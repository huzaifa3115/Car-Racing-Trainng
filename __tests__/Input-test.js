/**
 * @format
 */

import 'react-native';
import React from 'react';
import {Input} from '../src/Components';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {render, cleanup} from 'react-native-testing-library';

afterEach(cleanup);

describe('<Input />', () => {
  it('should match snapshot', () => {
    const rendered = render(<Input label={'name'} />).toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
