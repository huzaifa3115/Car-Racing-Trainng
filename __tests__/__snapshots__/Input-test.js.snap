// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
<TextInput
  allowFontScaling={true}
  label="name"
  rejectResponderTermination={true}
  selectionColor="#428AF8"
  style={
    Array [
      Object {
        "borderBottomWidth": 0.5,
        "borderColor": "#BEBEBE",
        "height": 40,
        "marginBottom": 20,
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
